### PR TITLE
Doc: correct fields, link updates.

### PR DESCRIPTION
**Pull request by kaovilai**
Remove name field from SnapshotLocation and BackupLocation.
Add bucket to BackupLocation property

### CODE REVIEW

File: docs/API_ref.md [hunk 1]
```diff
@@ -1,69 +1,69 @@
...
 |----------|---------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------|
-| name     | [metav1.ObjectMeta](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta)                                                                                |                                                                                                |
-| velero   | [*velero.BackupStorageLocationSpec](https://velero.io/docs/v1.9/api-types/backupstoragelocation/) | Location to store volume snapshots. For further details, see  [here] ( config/bsl_and_vsl.md). |
+| velero   | [*velero.BackupStorageLocationSpec](https://velero.io/docs/v1.9/api-types/backupstoragelocation/) | Location to store volume snapshots. For further details, see  [here](config/bsl_and_vsl.md). |
+| bucket   | [*CloudStorageLocation](https://pkg.go.dev/github.com/openshift/oadp-operator@master/api/v1alpha1#CloudStorageLocation) | Location to store volume snapshots. For further details, see  [here](config/bsl_and_vsl.md). |
```
Comment (kaovilai) [inline docs/API_ref.md:22]: Notable change 1

[the remark this answers is not in the document]
Comment (kaovilai) [reply docs/API_ref.md:22]: gotta fix description for bucket

File: docs/API_ref.md [hunk 2]
```diff
@@ -1,69 +1,69 @@
...
 |----------|---------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------|
-| name     | [metav1.ObjectMeta](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta)                                                                                |                                                                                                |
-| velero   | [*velero.BackupStorageLocationSpec](https://velero.io/docs/v1.9/api-types/backupstoragelocation/) | Location to store volume snapshots. For further details, see  [here] ( config/bsl_and_vsl.md). |
+| velero   | [*velero.BackupStorageLocationSpec](https://velero.io/docs/v1.9/api-types/backupstoragelocation/) | Location to store volume snapshots. For further details, see  [here](config/bsl_and_vsl.md). |
+| bucket   | [*CloudStorageLocation](https://pkg.go.dev/github.com/openshift/oadp-operator@master/api/v1alpha1#CloudStorageLocation) | [Tech Preview] Automate creation of bucket at some cloud storage providers for use as backup storage location |
```
Comment (kaovilai) [inline docs/API_ref.md:22]: Notable change 1

File: docs/API_ref.md [hunk 3]
```diff
@@ -1,69 +1,69 @@
...
 |----------|---------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------|
-| name     | [metav1.ObjectMeta](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta)                                                                                |                                                                                                |
-| velero   | [*velero.BackupStorageLocationSpec](https://velero.io/docs/v1.9/api-types/backupstoragelocation/) | Location to store volume snapshots. For further details, see  [here] ( config/bsl_and_vsl.md). |
+| velero   | [*velero.BackupStorageLocationSpec](https://velero.io/docs/v1.9/api-types/backupstoragelocation/) | Location to store volume snapshots. For further details, see  [here](config/bsl_and_vsl.md). |
+| bucket   | [*CloudStorageLocation](https://pkg.go.dev/github.com/openshift/oadp-operator@master/api/v1alpha1#CloudStorageLocation) | [Tech Preview] Automates creation of bucket at some cloud storage providers for use as a backup storage location |
```
Comment (kaovilai) [inline docs/API_ref.md:22]: Notable change 1

**Review by shubham-pampattiwar — APPROVED** [review]

File: docs/API_ref.md [hunk 4]
```diff
@@ -1,69 +1,69 @@
...
 
 | Property | Type                                                                                              | Description                                                                                    |
 |----------|---------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------|
-| name     | [metav1.ObjectMeta](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta)                                                                                |                                                                                                |
-| velero   | [*velero.BackupStorageLocationSpec](https://velero.io/docs/v1.9/api-types/backupstoragelocation/) | Location to store volume snapshots. For further details, see  [here] ( config/bsl_and_vsl.md). |
+| velero   | [*velero.BackupStorageLocationSpec](https://velero.io/docs/v1.9/api-types/backupstoragelocation/) | Location to store volume snapshots. For further details, see  [here](config/bsl_and_vsl.md). |
```
Comment (eemcmullan) [inline docs/API_ref.md:21]: Location to store backup objects?

**Review by eemcmullan — CHANGES_REQUESTED** [review]

**Review by dymurray — APPROVED** [review]

File: docs/API_ref.md (hunk 4 above)
Comment (kaovilai) [reply docs/API_ref.md:21]: thx

**Review by eemcmullan — APPROVED** [review]

**Pull request merged**